larsim/EventGenerator/TextFileGen_module.cc

```diff
@@ -94,17 +97,19 @@ class evgen::TextFileGen : public art::EDProducer {
 //------------------------------------------------------------------------------
 evgen::TextFileGen::TextFileGen(fhicl::ParameterSet const & p)
   : EDProducer{p}
+  , fOffset{p.get<unsigned long int>("Offset",0)}
```

Comment:
do you need the default value here?  Default values do make it harder to catch fcl mistakes...

Reply:
that makes it invisible to people who don't want this feature, i.e. people don't need to update their .fcl files if they don't want to.

Reply:
Is there a base FCL to update as that could absorb the change?  Otherwise I might argue that any new parameter is something an end user may want to know about.

Reply:
@therealendrju, are there any circumstances in which setting the offset value to something other than 0 is *necessary* to get the correct result?  If so, removing the C++ default of 0 (@absolution1's suggestion) may be the best option.

Reply:
> @therealendrju, are there any circumstances in which setting the offset value to something other than 0 is _necessary_ to get the correct result? If so, removing the C++ default of 0 (@absolution1's suggestion) may be the best option.

@therealendrju - comments please?

Reply:
Hi, sorry. Yes, the value is what steers the behaviour of the module, so if you want to use the new functionality you need to set it to something different than zero. I'll remove the default value and uncomment the .fcl file to provide the zero there.